ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/OperationPool.java

```diff
@@ -47,10 +45,12 @@ public OperationPool(Class<T> clazz, OperationStateTransitionValidator<T> operat
 
   public SSZList<T> getItemsForBlock(BeaconState stateAtBlockSlot) {
     SSZMutableList<T> itemsToPutInBlock =
-        SSZList.createMutable(clazz, maxNumberOfElementsInBlock.get(clazz));
+        SSZList.createMutable(clazz, MAX_NUMBER_OF_ELEMENTS_IN_BLOCK.get(clazz));
+    // Note that iterating through all items does not affect their access time so we are effectively
+    // evicting the oldest entries when the size is exceeded as we only ever access via iteration.
```

Comment:
is it worth investigating if we can explicitly make a call to update the accessed time of what we actually want to use?

Reply:
We used to deliberately use evict oldest policy here so this is actually what we wanted.